enacts/onset/maproom.py

```diff
@@ -599,6 +599,143 @@ def cess_plots(
         return cess_date_graph, cdf_graph, tab_style
 
 
+@APP.callback(
+    Output("length_plot", "figure"),
+    Output("length_prob_exc", "figure"),
+    Output("length_dbct","tab_style"),
+    Input("loc_marker", "position"),
+    Input("search_start_day", "value"),
+    Input("search_start_month", "value"),
+    Input("search_days", "value"),
+    Input("wet_threshold", "value"),
+    Input("running_days", "value"),
+    Input("running_total", "value"),
+    Input("min_rainy_days", "value"),
+    Input("dry_days", "value"),
+    Input("dry_spell", "value"),
+    Input("cess_start_day", "value"),
+    Input("cess_start_month", "value"),
+    Input("cess_search_days", "value"),
+    Input("cess_soil_moisture", "value"),
+    Input("cess_dry_spell", "value"),
+)
+def length_plots(
+    marker_pos,
+    search_start_day,
+    search_start_month,
+    search_days,
+    wet_threshold,
+    running_days,
+    running_total,
+    min_rainy_days,
+    dry_days,
+    dry_spell,
+    cess_start_day,
+    cess_start_month,
+    cess_search_days,
+    cess_soil_moisture,
+    cess_dry_spell,
+):
+    if not CONFIG["ison_cess_date_hist"]:
+        tab_style = {"display": "none"}
+        return {}, {}, tab_style
+    else:
+        tab_style = {}
+        lat = marker_pos[0]
+        lng = marker_pos[1]
+        try:
+            precip = pingrid.sel_snap(rr_mrg.precip, lat, lng)
+            isnan = np.isnan(precip).sum().sum()
```

Comment:
Why sum twice?

Reply:
I feel that question has been asked before. I feel like the answer could be something about doing a squeeze but for some reason it's easier/better this way; and so that it helps doing more straightforwarly the isnan > 0 test that follows. But I could be totally wrong.
This is also legacy code from cess_ case and I had the feeling it originally came from Kevin but I actually found it in this [PR](https://github.com/iridl/python-maprooms/pull/9) (see Oct 19th comment) from @drewmresnick even though this discussion doesn't argue specifically about using sum twice.

Reply:
Yes, I remember asking, but I don't remember getting a satisfactory answer. Could you just try deleting one of them and see what happens?

Reply:
It doesn't seem to make a difference and the last commit removes it. And we didn't keep the double sum in the flex forecast maproom for same purpose.

However, the unchanged behavior reveals unsatisfactory: if I make choices that make it impossible to find an onset date ever, I get the previous graph (ie graph remains unchanged) instead of an error graph... We are only testing whether there is precip data in the first place.

It looks like we need a test where data to plot is all missing, unless it's some dash option to trigger so that it doesn't keep the old graph.

Reply:
Indeed, the KeyError in line 653 and the TypeError in line 672 and 688 should all be generalized to Exception. We should always display an error message rather than continuing to display whatever happened to be there previously.

I think the reason we used specific exception types is because the error messages are specific. The easiest fix is to make the error messages less specific ("Something went wrong" or "Error" instead of "Grid box out of data domain"). A more helpful version could look something like
```
except Exception as e:
    if isinstance(e, KeyError):
        msg = "Grid box out of data domain"
    else:
        msg = "Internal error"
    error_fig = pingrid.error_fig(error_msg=msg)
    ....
```

Reply:
Alright, I left the checks on location clicks alone and added layers of checks to return an error fig if what feeds the graph is all nan.